zebra-test/src/command.rs

```diff
@@ -399,94 +399,178 @@ impl<T> TestOutput<T> {
         Ok(self)
     }
 
-    #[instrument(skip(self))]
-    pub fn stdout_contains(&self, regex: &str) -> Result<&Self> {
-        let re = regex::Regex::new(regex)?;
-        let stdout = String::from_utf8_lossy(&self.output.stdout);
+    /// Tests if the output of a command matches the given predicate.
+    ///
+    /// `output` is typically `self.output.stdout` or `self.output.stderr`.
```

Comment:
```suggestion
    /// The supplied `f` does the actual checking, can be any closure or function that returns a `bool`.
    /// `output` is typically `self.output.stdout` or `self.output.stderr`.
```

Reply:
I'd prefer to match the wording of `Iterator::filter` here:
https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter

But the function type is also over-constrained (it can be `FnOnce`), so I'm just going to push a fix commit.

Reply:
See commit 485721c1, which updates the generic name, argument names, and doc comments.